pkg/minishift/openshift/service.go

```diff
@@ -238,6 +220,60 @@ func filterAndUpdateServiceSpecs(routeSpec map[string][]ServiceWeight, namespace
 	return serviceSpecs
 }
 
+func serviceHasNodePort(serviceNodePort map[string]string, name string) bool {
+	_, ok := serviceNodePort[name]
+	return ok
+}
+
+func addToServiceSpecMap(serviceSpecMap map[string]*ServiceSpec, url string, service ServiceWeight, namespace string, np string) {
+	if !serviceInServiceSpecMap(serviceSpecMap, service.Name) {
+		addSpecToServiceSpecMap(serviceSpecMap, service, url, namespace, np)
+	} else {
+		appendUrlToServiceSpec(serviceSpecMap, service, url)
+		appendWeightToServiceSpec(serviceSpecMap, service.Name, service.Weight)
+	}
+}
+
+//serviceInServiceSpecMap returns true if the service is present in serviceSpecMap
+func serviceInServiceSpecMap(serviceSpecMap map[string]*ServiceSpec, name string) bool {
+	_, ok := serviceSpecMap[name]
+	return ok
+}
+
+func appendUrlToServiceSpec(serviceSpecMap map[string]*ServiceSpec, service ServiceWeight, url string) {
+	if service.Tls != "" {
```

Comment:
In the next part you use:

```diff
+	const (
+		http  = "http://%s"
+		https = "https://%s"
+	)
```

why not re-use this?

Reply:
Yeah.. I should re-use this...